### ISSUE TITLE
require strings to obviously be strings. maybe.

### ISSUE THREAD

**Issue by colemickens** [nushell/nushell#9336]
### Related problem

I am getting tripped up multiple times a day by one single thing -- unquoted strings being treated as strings.

I really wish that nushell didn't coerce anything it doesn't recognize to a string.

```
let success = true
...
if success {

}
```

gives a string error. Passing `gcrootsdir` to a external-command just silently, when I'd really meant `$gcrootsdir`, which meant I'd silently been writing to the completely wrong directory.

### Describe the solution you'd like

1. Nushell does not coerce unquoted strings to strings.

Granted, as I type that, I'm not really sure how that would be handled, given that random flags and arguments to external commands are indeed generally unquoted strings.

**Comment by fdncred** [comment]
success here is not a string
```
> let success = true
> $success | describe
bool
```

When you declare variables you need to reference them with `$variable_name` like most scripting languages. Maybe I'm not understanding?

**Comment by colemickens** [comment]
Right, but `if success { ... }` gives a "cannot convert string to bool" error. Which I've stared at a few times and gone... "no, it's not".

Maybe it would just be enough to have hints on errors. Like:

"Cannot convert string to bool (did you mean `$success` instead of `success`"

Given the external commands and "you're just doing it wrong", maybe there's no bug/feature here. But I've had this thought a few times and wanted to bring it up at least.

**Comment by fdncred** [comment]
someone may be able to hook up a 'did you mean' to this. i'm not sure how to do that myself.
